sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/AzureBlobDataFeedSource.cs

```diff
@@ -58,16 +58,42 @@ public AzureBlobDataFeedSource(string connectionString, string container, string
             BlobTemplate = blobTemplate;
         }
 
-        internal AzureBlobDataFeedSource(AzureBlobParameter parameter)
+        internal AzureBlobDataFeedSource(AzureBlobParameter parameter, AuthenticationTypeEnum? authentication)
```

Comment:
In each data source we're adding a `SetAuthentication` method (transform service => public), and a `GetAuthenticationTypeEnum` method (transform public => service).